SYCL/Plugin/level_zero_queue_profiling.cpp

```diff
@@ -0,0 +1,82 @@
+// REQUIRES: gpu, level_zero
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// RUN: env ZE_DEBUG=-1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --check-prefixes=WITHOUT %s
+// RUN: env ZE_DEBUG=-1 %GPU_RUN_PLACEHOLDER %t.out profile 2>&1 | FileCheck --check-prefixes=WITH %s
+
+// Test case adapted from the SYCL version of Rodinia benchmark hotspot.
+
+// Check the expected output when queue::enable_profiling is not specified
+//
+// WITHOUT: ze_event_pool_desc_t flags set to: 1
+// WITHOUT: terminate called after throwing an instance of 'cl::sycl::runtime_error'
+// WITHOUT:   what():  Native API failed. Native API returns: -7 (CL_PROFILING_INFO_NOT_AVAILABLE)
+
+// Check the expected output when queue::enable_profiling is specified
+//
+// WITH: ze_event_pool_desc_t flags set to: 5
+// WITH: Kernel time:
+// WITH: Device kernel time:
+// WITH: Device offloading time:
+
+#include <CL/sycl.hpp>
+using namespace cl::sycl;
+#include <sys/time.h>
+
+// wrapper for gettimeofday
+long long get_time() {
```

Comment:
get_time() usages look irrelevant to the test. Wouldn't it be enough to test only profiling functionality (Device kernel time)?

Reply:
Yes, it would. The reasons that I prefer to leave get_time() in are: 1) it came from a customer so reflects real case better, and 2) it may be helpful for debugging profiling issues, i.e. comparing host clock to that reported by the device.

Reply:
On the other hand, it makes the test Linux-only. I removed the host timers from the test.